### PR TITLE
fix: preserve source-scoped producer handoff

### LINKED ISSUE #62: Skill feedback: gh-address-cr - Empty synced findings are accepted by `findings --sync --source` but not cons...
# Skill feedback: gh-address-cr - Empty synced findings are accepted by `findings --sync --source` but not cons...

## Skill
- skill: `gh-address-cr`
- version_or_path: `gh-address-cr 2.5.1; installed skill [redacted-path]`
- target_kind: `repo`

## Problem
Empty synced findings are accepted by `findings --sync --source` but not consumed by the next plain `review` command. In practice, `gh-address-cr findings --input - --sync --source <producer>` can report `PASSED` for an empty findings payload, but a following `gh-address-cr review <repo> <pr>` still waits for external review handoff.

## Expected Behavior
After an external producer submits `[]` through `findings --input - --sync --source code-review` and the command reports `PASSED`, the next plain `review <repo> <pr>` should not return `WAITING_FOR_EXTERNAL_REVIEW`. Either write canonical `incoming-findings.json` as `[]`, teach `review` to consume the source-scoped synced findings state, or make the `findings` next action explicit that the operator must pass canonical input to `review`.

## Observed Behavior
On PR `RbBtSn0w/apple-skills#19`, `printf '[]\n' | gh-address-cr findings RbBtSn0w/apple-skills 19 --input - --sync --source code-review` returned `status: PASSED` with zero open local findings. Re-running `gh-address-cr review RbBtSn0w/apple-skills 19` still returned `status: WAITING_FOR_EXTERNAL_REVIEW`. The PR cache showed `incoming-findings.json` and `incoming-findings.md` as zero-byte files. The workaround was `printf '[]\n' | gh-address-cr review RbBtSn0w/apple-skills 19 --input -`, which advanced the session into GitHub thread handling.

## Evidence
Runtime version: `gh-address-cr 2.5.1`. `findings --input - --sync --source code-review` output included `status: PASSED`, `reason_code: PASSED`, `open_local_findings_count: 0`, and `unresolved_github_threads_count: 0`. A subsequent plain `review RbBtSn0w/apple-skills 19` output included `status: WAITING_FOR_EXTERNAL_REVIEW`, `reason_code: WAITING_FOR_EXTERNAL_REVIEW`, and `next_action: Provide findings JSON in incoming-findings.json or fixed finding blocks in incoming-findings.md`. `review --input -` with the same `[]` payload then advanced to `status: BLOCKED`, `reason_code: BLOCKING_ITEMS_REMAIN`, proving the empty findings payload was valid when passed directly to `review`.

## Repro Steps
- Start a PR session where `gh-address-cr review <repo> <pr>` returns `WAITING_FOR_EXTERNAL_REVIEW`.
- Run `printf '[]\n' | gh-address-cr findings <repo> <pr> --input - --sync --source code-review`.
- Observe that `findings` reports `PASSED`.
- Run `gh-address-cr review <repo> <pr>` without `--input`.
- Observe that it still returns `WAITING_FOR_EXTERNAL_REVIEW` instead of consuming the accepted empty producer result.

## Impact
medium

## Repeatability
likely_recurring

## Proposed Validation
Add a regression test or CLI integration fixture for the empty external producer path: invoke `findings --input - --sync --source code-review` with `[]`, then invoke plain `review <repo> <pr>` and assert it does not return `WAITING_FOR_EXTERNAL_REVIEW`. If the intended contract is that `findings` is not consumed by plain `review`, assert that the `findings` machine summary has a `next_action` telling the operator to rerun `review --input -` or write canonical `incoming-findings.json` explicitly.

## Non-Goal
Do not skip external review, weaken GitHub thread evidence, or bypass `final-gate`. This feedback is only about making the external findings handoff contract consistent and self-explanatory.

## Intake Triage
- decision: `accept`
- reasons:
  - ready for maintainer review

## Maintainer Workflow
- owning_skill: `gh-address-cr`
- owning_skill_path: `gh-address-cr 2.5.1; installed skill [redacted-path]`
- suggested_labels: `skill-feedback`, `skill:gh-address-cr`, `impact:medium`, `repeatability:likely_recurring`
- close_gate: keep this issue open until a separate PR updates the owning skill and its validation passes.
- non_goal: intake does not edit the skill, open the implementation PR, merge the PR, or close this issue automatically.

### PR DESCRIPTION
**Pull request by RbBtSn0w**
## Summary
- Add a source-scoped producer result marker under `session.handoff.producer_results` whenever findings input is successfully ingested, including explicit empty `[]` payloads.
- Teach plain `review` to continue the existing PR session when a producer result has already been submitted through `findings --input <path>|- --source <producer>`.
- Update docs, skill payload, plugin payload, and regression coverage for the command-to-command handoff contract.

## Requirement Breakdown
- `findings --input - --sync --source code-review` with `[]` must be treated as an explicit empty producer result.
- A following plain `review <repo> <pr>` must not return `WAITING_FOR_EXTERNAL_REVIEW` solely because canonical handoff files are still empty.
- Non-empty findings ingested through `findings` must likewise be visible to a following plain `review`, which should continue to local finding handling.
- Empty stdin remains invalid; only parsed findings JSON, including `[]`, records a producer result.

## Validation
- `PYENV_VERSION=3.10.19 /opt/homebrew/bin/pyenv exec ruff check src tests scripts/build_plugin_payload.py`
- `PYENV_VERSION=3.10.19 /opt/homebrew/bin/pyenv exec python -m unittest discover -s tests`
- `PYENV_VERSION=3.10.19 /opt/homebrew/bin/pyenv exec python -m gh_address_cr --help`
- `PYENV_VERSION=3.10.19 /opt/homebrew/bin/pyenv exec python skill/scripts/cli.py --help`
- `PYENV_VERSION=3.10.19 /opt/homebrew/bin/pyenv exec python -m gh_address_cr agent manifest`
- `PYENV_VERSION=3.10.19 /opt/homebrew/bin/pyenv exec python scripts/build_plugin_payload.py --check`
- Manual reproduction of issue #62 path: initial `review` returned `WAITING_FOR_EXTERNAL_REVIEW`, `findings --input - --sync --source code-review` with `[]` returned `PASSED` and recorded `producer_results.code-review`, and the next plain `review` returned `PASSED`.

Fixes #62

### CODE REVIEW

File: src/gh_address_cr/cli.py [hunk 1]
```diff
@@ -789,6 +803,9 @@ def preflight_high_level(args: argparse.Namespace) -> int | None:
             if handoff_sha256:
                 args.args.extend(["--handoff-sha256", handoff_sha256])
             return None
+        if has_submitted_producer_result(repo, pr_number):
+            args.review_continue_without_input = True
+            return None
```
[the remark this answers is not in the document]
Comment (RbBtSn0w) [reply src/gh_address_cr/cli.py:808]: Fixed in `5a059a0`.

Severity: `P2`

What I changed:
- `src/gh_address_cr/cli.py`: Removed unused native review handoff flag assignments.

Why this addresses the CR:
- The review command now continues directly when a handoff was already consumed or a source-scoped producer result exists; the legacy rewrite flag is no longer mutated on the native path.
- Medium-severity path validated and aligned with expected workflow.

Validation:
- `PYENV_VERSION=3.10.19 /opt/homebrew/bin/pyenv exec ruff check src tests scripts/build_plugin_payload.py && PYENV_VERSION=3.10.19 /opt/homebrew/bin/pyenv exec python -m unittest discover -s tests`
- Result: passed

File: src/gh_address_cr/cli.py [hunk 2]
```diff
@@ -962,8 +992,20 @@ def _ingest_native_findings(
...
+        canonical_findings_payload(findings).encode("utf-8")
+    ).hexdigest()
+    producer_results[source] = {
+        "status": "submitted",
+        "source": source,
+        "findings_count": len(findings),
+        "payload_sha256": payload_sha256,
+        "sync_enabled": bool(sync),
+        "submitted_at": now,
+    }
```
[the remark this answers is not in the document]
Comment (RbBtSn0w) [reply src/gh_address_cr/cli.py]: Fixed in `5a059a0`.

Severity: `P2`

What I changed:
- `src/gh_address_cr/core/handoff.py`: Centralized source-scoped producer result recording.
- `src/gh_address_cr/cli.py`: Centralized source-scoped producer result recording.
- `src/gh_address_cr/core/session_engine.py`: Centralized source-scoped producer result recording.

Why this addresses the CR:
- Both the native CLI and session engine now call gh_address_cr.core.handoff.record_producer_result, so the submitted marker shape and payload hash behavior have one owner instead of duplicated dictionaries.
- Medium-severity path validated and aligned with expected workflow.

Validation:
- `PYENV_VERSION=3.10.19 /opt/homebrew/bin/pyenv exec ruff check src tests scripts/build_plugin_payload.py && PYENV_VERSION=3.10.19 /opt/homebrew/bin/pyenv exec python -m unittest discover -s tests`
- Result: passed

File: src/gh_address_cr/core/session_engine.py [hunk 3]
```diff
@@ -204,8 +205,40 @@ def ensure_loop_state(session: dict):
...
+        handoff = {}
+        session["handoff"] = handoff
     handoff.setdefault("last_consumed_sha256", None)
+    producer_results = handoff.get("producer_results")
+    if not isinstance(producer_results, dict):
+        handoff["producer_results"] = {}
+
+
+def canonical_findings_payload(findings: list[dict]) -> str:
+    return json.dumps(findings, sort_keys=True, separators=(",", ":"))
```
[the remark this answers is not in the document]
Comment (RbBtSn0w) [reply src/gh_address_cr/core/session_engine.py]: Fixed in `5a059a0`.

Severity: `P2`

What I changed:
- `src/gh_address_cr/intake/findings.py`: Moved canonical findings payload hashing to shared intake code.
- `src/gh_address_cr/core/handoff.py`: Moved canonical findings payload hashing to shared intake code.
- `src/gh_address_cr/cli.py`: Moved canonical findings payload hashing to shared intake code.
- `src/gh_address_cr/core/session_engine.py`: Moved canonical findings payload hashing to shared intake code.

Why this addresses the CR:
- canonical_findings_payload now lives in gh_address_cr.intake.findings and is consumed by both review handoff normalization and producer result recording, keeping hash semantics consistent.
- Medium-severity path validated and aligned with expected workflow.

Validation:
- `PYENV_VERSION=3.10.19 /opt/homebrew/bin/pyenv exec ruff check src tests scripts/build_plugin_payload.py && PYENV_VERSION=3.10.19 /opt/homebrew/bin/pyenv exec python -m unittest discover -s tests`
- Result: passed

File: src/gh_address_cr/cli.py [hunk 4]
```diff
@@ -962,8 +992,20 @@ def _ingest_native_findings(
                 item["handled"] = True
                 item["handled_at"] = now
                 item["updated_at"] = now
+    handoff = _ensure_handoff_state(session)
+    producer_results = handoff["producer_results"]
+    payload_sha256 = handoff_sha256 or hashlib.sha256(
+        canonical_findings_payload(findings).encode("utf-8")
+    ).hexdigest()
```
[the remark this answers is not in the document]
Comment (RbBtSn0w) [reply src/gh_address_cr/cli.py]: Fixed in `5a059a0`.

Severity: `P2`

What I changed:
- `src/gh_address_cr/cli.py`: Rejected blank native findings input before producer marker writes.
- `src/gh_address_cr/intake/findings.py`: Rejected blank native findings input before producer marker writes.
- `tests/test_python_wrappers.py`: Rejected blank native findings input before producer marker writes.

Why this addresses the CR:
- _ingest_native_findings now fails fast on empty raw input, so only explicit JSON such as [] can create a submitted producer result; the regression test proves producer_results is left untouched for blank stdin.
- Medium-severity path validated and aligned with expected workflow.

Validation:
- `PYENV_VERSION=3.10.19 /opt/homebrew/bin/pyenv exec ruff check src tests scripts/build_plugin_payload.py && PYENV_VERSION=3.10.19 /opt/homebrew/bin/pyenv exec python -m unittest discover -s tests`
- Result: passed

File: src/gh_address_cr/core/session_engine.py [hunk 5]
```diff
@@ -700,6 +733,13 @@ def cmd_ingest_local(args):
             synced += 1
     if args.handoff_sha256:
         session["handoff"]["last_consumed_sha256"] = args.handoff_sha256
+    record_producer_result(
+        session,
+        source=args.source,
+        findings=findings,
+        sync_enabled=bool(args.sync),
```
[the remark this answers is not in the document]
Comment (RbBtSn0w) [reply src/gh_address_cr/core/session_engine.py:707]: Fixed in `5a059a0`.

Severity: `P2`

What I changed:
- `src/gh_address_cr/core/session_engine.py`: Rejected blank ingest-local stdin before recording producer results.
- `src/gh_address_cr/core/handoff.py`: Rejected blank ingest-local stdin before recording producer results.
- `tests/test_session_engine_cli.py`: Rejected blank ingest-local stdin before recording producer results.

Why this addresses the CR:
- read_records_from_stdin now has an explicit require_payload gate used by ingest-local, preserving [] as an explicit empty result while preventing blank stdin from closing source-scoped items or writing submitted markers.
- Medium-severity path validated and aligned with expected workflow.

Validation:
- `PYENV_VERSION=3.10.19 /opt/homebrew/bin/pyenv exec ruff check src tests scripts/build_plugin_payload.py && PYENV_VERSION=3.10.19 /opt/homebrew/bin/pyenv exec python -m unittest discover -s tests`
- Result: passed

File: docs/cli-reference.md [hunk 6]
```diff
@@ -306,10 +312,10 @@ Prompt patterns:
 写出 `producer-request.md`、`incoming-findings.json`、`incoming-findings.md`。
 
 如果你自己就是外部 review producer，就在当前任务里直接生成 findings JSON，
-写入 `incoming-findings.json`；或者生成固定格式的 `finding` blocks`，
+写入 `incoming-findings.json`，或者通过 `findings --input - --source <producer>` 交给同一 PR session；或者生成固定格式的 `finding` blocks`，
```
[the remark this answers is not in the document]
Comment (RbBtSn0w) [reply docs/cli-reference.md]: Fixed in `5a059a0`.

Severity: `P2`

What I changed:
- `docs/cli-reference.md`: Removed malformed inline-code backtick in CLI docs.

Why this addresses the CR:
- The prompt pattern now renders  blocks correctly without the stray trailing backtick.
- Medium-severity path validated and aligned with expected workflow.

Validation:
- `PYENV_VERSION=3.10.19 /opt/homebrew/bin/pyenv exec ruff check src tests scripts/build_plugin_payload.py && PYENV_VERSION=3.10.19 /opt/homebrew/bin/pyenv exec python -m unittest discover -s tests`
- Result: passed

File: src/gh_address_cr/cli.py [hunk 7]
```diff
@@ -962,8 +992,20 @@ def _ingest_native_findings(
...
+    producer_results[source] = {
+        "status": "submitted",
+        "source": source,
+        "findings_count": len(findings),
```
[the remark this answers is not in the document]
Comment (RbBtSn0w) [reply src/gh_address_cr/cli.py]: Fixed in `5a059a0`.

Severity: `P2`

What I changed:
- `src/gh_address_cr/cli.py`: Gated producer result submission on explicit findings payload presence.
- `src/gh_address_cr/intake/findings.py`: Gated producer result submission on explicit findings payload presence.
- `tests/test_python_wrappers.py`: Gated producer result submission on explicit findings payload presence.

Why this addresses the CR:
- A blank producer pipe now returns INVALID_FINDINGS_INPUT before handoff.producer_results is mutated, while explicit [] remains accepted and covered by the existing empty-result regression.
- Medium-severity path validated and aligned with expected workflow.

Validation:
- `PYENV_VERSION=3.10.19 /opt/homebrew/bin/pyenv exec ruff check src tests scripts/build_plugin_payload.py && PYENV_VERSION=3.10.19 /opt/homebrew/bin/pyenv exec python -m unittest discover -s tests`
- Result: passed
